packages/gatsby-plugin-google-analytics/src/gatsby-browser.js

```diff
@@ -1,7 +1,12 @@
-exports.onRouteUpdate = function(location) {
+var lastPath = null
+exports.onRouteUpdate = function(newRoute) {
   // Don't track while developing.
   if (process.env.NODE_ENV === `production`) {
-    ga(`set`, `page`, location.pathname)
-    ga(`send`, `pageview`)
+    var newPath = newRoute.pathname || newRoute.location.pathname
```

Comment:
Ugh, yeah this code is the cause :-( https://github.com/gatsbyjs/gatsby/blob/08bddd92b60bccd125b2c147da4ec896f48197c3/packages/gatsby/src/cache-dir/production-app.js#L82

Try changing that function so that it only attaches once.